File: com.systemincloud.examples.tasks.javatask/src/test/java/tasks/generator/GeneratorAtomicWith2SIn1AIn.java
package tasks.generator;

import com.systemincloud.modeler.tasks.javatask.api.InputPort;
import com.systemincloud.modeler.tasks.javatask.api.JavaTask;
import com.systemincloud.modeler.tasks.javatask.api.OutputPort;
import com.systemincloud.modeler.tasks.javatask.api.annotations.InputPortInfo;
import com.systemincloud.modeler.tasks.javatask.api.annotations.JavaTaskInfo;
import com.systemincloud.modeler.tasks.javatask.api.annotations.OutputPortInfo;
import com.systemincloud.modeler.tasks.javatask.api.data.Int32;
import com.systemincloud.modeler.tasks.javatask.api.data.Control;

@JavaTaskInfo(generator = true)
public class GeneratorAtomicWith2SIn1AIn extends JavaTask {

	@InputPortInfo(name = "In1", dataType = Int32.class)
	public InputPort in1;
	@InputPortInfo(name = "In2", dataType = Int32.class)
	public InputPort in2;
	@InputPortInfo(name = "In3", dataType = Control.class, asynchronus = true)
	public InputPort in3;
	@OutputPortInfo(name = "Out", dataType = Int32.class)
	public OutputPort out;

	private boolean on = false;
	
	@Override
	public void executeAsync(InputPort asynchIn) {
		this.on = !on;
	}

	@Override
	public void generate() {
		if(on) out.putData(new Int32(637343730));
		sleep(500);
	}

	@Override
	public void execute() {
		out.putData(new Int32(in1.getData(Int32.class).getValue() + in2.getData(Int32.class).getValue()));
	}

}

File: com.systemincloud.examples.tasks.pythontask/src/test/py/tasks/generator/GeneratorAtomicWith2SIn1AIn.py
from sicpythontask.PythonTaskInfo import PythonTaskInfo
from sicpythontask.PythonTask import PythonTask
from sicpythontask.InputPort import InputPort
from sicpythontask.OutputPort import OutputPort
from sicpythontask.data.Int32 import Int32
from sicpythontask.data.Control import Control

@PythonTaskInfo(generator=True)
class GeneratorAtomicWith2SIn1AIn(PythonTask):

    def __init_ports__(self):
        self.in1 = InputPort(name="in1", data_type=Int32)
        self.in2 = InputPort(name="in2", data_type=Int32)
        self.in3 = InputPort(name="in3", data_type=Control, asynchronous=True)
        self.out = OutputPort(name="out", data_type=Int32)

    def runner_start(self):
        self.on = True

    def execute_async(self, async_in):
        self.on = not self.on

    def generate(self):
        if self.on:
            self.out.put_data(Int32(637343730));
        self.sleep(500);

    def execute(self, grp):
        self.out.put_data(Int32(self.in1.get_data(Int32).values[0] + self.in2.get_data(Int32).values[0]));
